Lazy/Lazy/ILazy.cs

```diff
@@ -0,0 +1,20 @@
+﻿// <copyright file="ILazy.cs" company="Bulat Tsydendorzhiev">
+// Copyright (c) Bulat Tsydendorzhiev. All Rights Reserved.
+// Licensed under the MIT License. See LICENSE in the repository root for license information.
+// </copyright>
+
+namespace Lazy;
+
+/// <summary>
+/// An interface used for lazy evaluation.
+/// </summary>
+/// <typeparam name="T">The type of the value.</typeparam>
+public interface ILazy<T>
```

Comment:
```suggestion
public interface ILazy<out T>
```
Если есть возможность указать вариантность, лучше указать, это упростит пользование контейнером.